### PR TITLE
For 2.5: Proposal: Use Kotlin as the primary development language for…

### LINKED ISSUE #648: Proposal: Use Kotlin as the primary development language for Kiwix
#### Feature Description
Last year Koltin became an official language for Android Development. Since then, it has received enormous love from the community. A few advantages of using Kotlin over Java are:
1. Kotlin is a type and null-safe language.
2. Kotlin is concise and fun to write. It makes Android Development a lot faster.
3. It is interoperable with Java. We can call Kotlin code from Java code and vice-versa. 
4. Companies like Slack, Pinterest are already using it.
5. It is easy to learn. You can literally start writing Kotlin code after 2-3 days of [practice](https://kotlinlang.org/docs/tutorials/koans.html).
6. Google is also providing their support to Kotlin. A few months before, Google marked many function parameters with `@NonNull` annotation ([read more](https://proandroiddev.com/android-kotlin-and-api-27-2e6f3f893c2)). This was done to help Kotlin figure out the type of the parameters at compile time. Also, In February, Google released [android-ktx](https://github.com/android/android-ktx) library which contains very useful extension functions for commonly used android classes.
7. The [Kotlin Android Extensions](https://kotlinlang.org/docs/tutorials/android-plugin.html) plugin allows you to [replace butterknife](https://stackoverflow.com/a/46482618/7263373) library with its view binding abilities without annotating exposed fields!
8. Kotlin includes a lot of extension functions for the Collection class which lets you operate on these easily. For e.g., Kotlin Collections contains methods like `forEach()`, `filter()`, `findFirst()`, etc. which are only available for higher Android API's when using Java 8. 
The list doesn't end here. Kotlin's interoperability with Java lets you convert existing Java code one file at a time. 
@mhutti1 @kelson42 @rashiq Please consider this proposal.

### PR DESCRIPTION
**Pull request by macgills**
… Kiwix #648

Fixes #648 
Changes: update build scripts to support kotlin

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,14 +1,16 @@
 
 
 buildscript {
-  repositories {
+    ext.kotlin_version = '1.3.30'
+    repositories {
```
Comment (abdulwd) [inline build.gradle]: Use 2 space indents.

File: build.gradle [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
     jcenter()
       google()
   }
   dependencies {
-      classpath 'com.android.tools.build:gradle:3.1.0'
+      classpath 'com.android.tools.build:gradle:3.4.0'
+      classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
 
-    // NOTE: Do not place your application dependencies here; they belong
+      // NOTE: Do not place your application dependencies here; they belong
```
Comment (abdulwd) [inline build.gradle]: Don't make unnecessary changes.

File: build.gradle (hunk 1 above)
Comment (macgills) [reply build.gradle]: I set my Groovy formatting options to be 2 spaces and the formatting of the entire file was changed. Is the 2 spaces style a post 2.4 convention?
